packages/property-grid/src/components/PropertyGridWithUnifiedSelection.tsx

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as React from "react";
+
+import {
+  FilteringPropertyDataProvider,
+  IPropertyDataProvider,
+  PropertyCategory,
+  PropertyData,
+  PropertyDataChangeEvent,
+  PropertyDataFiltererBase,
+  VirtualizedPropertyGridWithDataProvider,
+  VirtualizedPropertyGridWithDataProviderProps
+} from "@bentley/ui-components";
+import { IPresentationPropertyDataProvider, usePropertyDataProviderWithUnifiedSelection } from "@bentley/presentation-components";
+import { FillCentered } from "@bentley/ui-core";
+import { IModelApp } from "@bentley/imodeljs-frontend";
+import { IDisposable } from "@bentley/bentleyjs-core";
+
+type CustomPropertyDataProvider<TPropertyData> = IDisposable & Omit<IPropertyDataProvider, "getData"> & { getData: () => Promise<TPropertyData> };
+class AutoExpandingPropertyDataProvider<TPropertyData extends PropertyData> implements IPropertyDataProvider, IDisposable {
+  public onDataChanged = new PropertyDataChangeEvent();
+  public constructor(private _wrapped: CustomPropertyDataProvider<TPropertyData>) {
+    this._wrapped.onDataChanged.addListener(() => this.onDataChanged.raiseEvent());
+  }
+  public dispose() { this._wrapped.dispose(); }
+  public async getData(): Promise<TPropertyData> {
+    function expandCategories(categories: PropertyCategory[]) {
+      categories.forEach((category: PropertyCategory) => {
+        category.expand = true;
+        if (category.childCategories)
+          expandCategories(category.childCategories);
+      });
+    }
+    const result = await this._wrapped.getData();
+    expandCategories(result.categories);
+    return result;
+  }
+}
+
+interface PropertyGridWithUnifiedSelectionProps extends VirtualizedPropertyGridWithDataProviderProps {
+  filterer: PropertyDataFiltererBase;
+}
+
+export function PropertyGridWithUnifiedSelection(props: PropertyGridWithUnifiedSelectionProps): JSX.Element {
+  const { isOverLimit /*, numSelectedElements*/ } = usePropertyDataProviderWithUnifiedSelection(
```

Comment:
is numSelectedElements not needed? If so, id remove that and the code block on L60-L66 thats commented out

Reply:
That return value has not been added to the latest package release, and so I had just commented it out to include it as necessary when the newer version comes out.

Reply:
is that coming out in 2.17? id personally rather not bump the major back to back, and would just hold this pr until the ver is out if thats cool with @jason-crow

Reply:
@aruniverse is does appear to be in the 2.17 release branch:
https://github.com/imodeljs/imodeljs/blob/release/2.17.x/presentation/components/src/presentation-components/propertygrid/UseUnifiedSelection.tsx

however, wouldn't this just be undefined until 2.17? in otherwords, couldn't you skirt the major bump by just checking 
if (numSelectedElements !== undefined && numSelectedElements === 0) 
and just keep it uncommented with a comment saying 'numSelectedElements will be undefined why used by versions older than 2.17 of presentation-components'

Reply:
i like that! as long as @calebmshafer  is in agreement, we can do this

Reply:
will probably want to cast it like this to make it not throw a lint error:
as UsePropertyDataProviderWithUnifiedSelectionResult & { numSelectedElements: number | undefined }

Reply:
Do we intend to drop the minimum version again and then do what @jason-crow suggested?  If so, I agree I like it to avoid a major version bump.

Reply:
Talking to Utkarsh, sounded like what he needs in this pr is the `usePropertyDataProviderWithUnifiedSelection` hook added in [2.6.0](https://github.com/imodeljs/imodeljs/blob/383386291a45f0b3435bcb44e83840ee02f31f5a/presentation/components/CHANGELOG.md#updates-9) and `FilteringPropertyDataProvider` added in [2.7.0](https://github.com/imodeljs/imodeljs/commit/5c325007bce78d8ce0ce74e505f113b8b5ce31cb#diff-45915886c77b97687147d5b46c5258952c9999ab40d5a6796610dfc5c51a0f2e) _I think_, so we'll probably have to bump to at least ^2.7.0